core/core.go

```diff
@@ -104,7 +105,7 @@ type IpfsNode struct {
 
 	IpnsFs *ipnsfs.Filesystem
 
-	ctxgroup.ContextGroup
+	goprocess.Process
```

Comment:
what i've been doing with goprocess is to make this a variable, not embedded type:

```go
type Object struct {
  proc goprocess.Process
}
```

And then have a function that returns the process.

```go
func (o Object) Process() goprocess.Process {
  return o.proc
}
```

the reason is that the Process interface is pretty large and embedding it dumps a bunch of objects on the user. maybe move to that? sorry for not describing it earlier.